napalm/base/base.py

```diff
@@ -218,9 +218,33 @@ def compare_config(self):
         """
         raise NotImplementedError
 
-    def commit_config(self, message=""):
+    def commit_config(self, message="", confirm_timeout=None):
         """
         Commits the changes requested by the method load_replace_candidate or load_merge_candidate.
+
+        :param message: Optional - configuration session commit message
+        :type message: str
+        :param confirm_timeout: Optional - The # of seconds before the configuration must be
+        confirmed via the commit_confirm method before OS reverts back to the original configuration
+        :type confirm_timeout: int
+        """
+        raise NotImplementedError
+
+    def commit_confirm(self):
```

Comment:
Yeah, JUNOS and IOSXR are most similar in those regards, but it varies by platform:
```
IOS: configure replace time 10; configure replace confirm
IOSXR: commit confirm 600; commit
NXOS: configure replace commit-timeout 600; configure replace commit
EOS: configuration session X timer 600; configuration session X confirm
JUNOS: commit confirmed 10; commit -or- commit check
```
since the order of operations was to call `commit_config` followed by `commit_confirm` it makes sense to me to prefix them the same vs shaping it from the OS's choices in CLI.  but i do understand your point re: `commit_config` followed by `confirm_commit` is what's being done

Reply:
Okay, I am open to changing this to `confirm_commit`...yeah, I see how it could be confusing so `confirm_commit` sounds good to me.

So operations would be:

```
commit_config(commit_confirm=True)
confirm_commit()
```

Reply:
:+1:

Reply:
> @rbcollins123 Let me know when the initial stub is ready and I will review.
> 
> Regards, Kirk

@ktbyers I rebased to get rid of that ancient commit and then tried to capture everything in the discussions above.  Take a look and let me know if you see anything I missed.

Reply:
done